spec/address_book_spec.rb

```diff
@@ -190,4 +192,53 @@ def check_entry(entry, expected_name, expected_number, expected_email)
       expect(entry).to be_nil
     end
   end
+
+  context '#iterative_search' do
+    it 'searches AddressBook for a non-existent entry' do
+      book.import_from_csv('entries.csv')
+      entry = book.iterative_search("Dan")
+      expect(entry).to be_nil
+    end
+
+    it 'searches AddressBook for Bill' do
+      book.import_from_csv('entries.csv')
+      entry = book.iterative_search('Bill')
+      expect(entry).to be_a Entry
+      check_entry(entry, 'Bill', '555-555-4854', 'bill@blocmail.com')
+    end
+
+    it 'searches AddressBook for Bob' do
+      book.import_from_csv('entries.csv')
+      entry = book.iterative_search('Bob')
+      expect(entry).to be_a Entry
+      check_entry(entry, 'Bob', '555-555-5415', 'bob@blocmail.com')
+    end
+
+    it 'searches AddressBook for Joe' do
+      book.import_from_csv('entries.csv')
+      entry = book.iterative_search('Joe')
+      expect(entry).to be_a Entry
+      check_entry(entry, 'Joe', '555-555-3660', 'joe@blocmail.com')
+    end
+
+    it 'searches AddressBook for Sally' do
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.